### PR TITLE
Mark contact candidates

### PR DESCRIPTION
**Pull request by srwopschall**
- This more clearly marks contact candidates emerging from the computational geometry with associated contact planes
- The contact candidate mark on the interface pairs is more clearly used throughout the code
- An old 'inContact' boolean on the interface pairs is deprecated in favor of marking 'in contact' on the contact plane
- The timestep calculation skips interface pairs that are NOT contact candidates, which is a bugfix.

### CODE REVIEW

File: src/tribol/mesh/CouplingScheme.cpp [hunk 1]
```diff
@@ -916,23 +916,23 @@ int CouplingScheme::apply( integer cycle, real t, real &dt )
...
      else
      {
-        pair.inContact = true;
+        pair.isContactCandidate = true;
         ++numActivePairs;
      }
      
      // update the InterfacePairs container on the coupling scheme 
-     // to reflect the change to "in-contact"
+     // to reflect the change to isContactCandidate = true
```
Comment (liu15) [inline src/tribol/mesh/CouplingScheme.cpp]: Couldn't this be set to false on 926?

Comment (srwopschall) [reply src/tribol/mesh/CouplingScheme.cpp]: Yes. The comment has been updated.

**Review by liu15 — APPROVED** [review]

File: src/tribol/geom/ContactPlane.cpp [hunk 2]
```diff
@@ -29,9 +29,9 @@ namespace tribol
 FaceGeomError CheckInterfacePair( InterfacePair& pair,
                                   ContactMethod const cMethod,
                                   ContactCase const TRIBOL_UNUSED_PARAM(cCase),
-                                  bool& inContact )
+                                  bool& isInteracting )
```
Comment (ebchin) [inline src/tribol/geom/ContactPlane.cpp:32]: I think the header needs to be updated too.  Also, a definition of an interacting face pair would be helpful.

File: src/tribol/interface/tribol.cpp [hunk 3]
```diff
@@ -947,14 +947,19 @@ void setInterfacePairs( integer couplingSchemeIndex,
       // since a user may specify interface pairs that aren't, nor should be in contact.
       InterfacePair pair { meshId1[i], pairType1[i], pairIndex1[i],
                            meshId2[i], pairType2[i], pairIndex2[i],
-                           false, i };
+                           true, i };
```
Comment (ebchin) [inline src/tribol/interface/tribol.cpp]: Why is this changed from false to true?  Should we use a constructor that doesn't set this if this is just a placeholder value?

File: src/tribol/mesh/CouplingScheme.cpp [hunk 4]
```diff
@@ -1372,7 +1382,7 @@ void CouplingScheme::computeCommonPlaneTimeStep(real &dt)
       real max_delta1 = proj_ratio * t1;
       real max_delta2 = proj_ratio * t2;
 
-      if (pair.inContact) // pair passes all geometric checks
+      if (cpMgr.m_inContact[cpID]) // pair passes all geometric checks
```
Comment (ebchin) [inline src/tribol/mesh/CouplingScheme.cpp]: Is the cpMgr the only place where a pair's status is stored now?

File: src/tribol/search/InterfacePairFinder.cpp [hunk 5]
```diff
@@ -509,16 +514,21 @@ class GridSearch
...
           {
+             pair.isContactCandidate = true;
              pair.pairId = k;
              contactPairs->addInterfacePair( pair );
              ++k;
           }
+          else
+          {
+             pair.isContactCandidate = false;
+          }
```
Comment (ebchin) [inline src/tribol/search/InterfacePairFinder.cpp]: Seems like this block is pretty similar to the above one (L435-449).  Can they be moved to a single function?  Maybe a new InterfacePair constructor?

**Review by ebchin — CHANGES_REQUESTED** [review]
Thanks @srwopschall, these are good changes that make the code clearer.  I left a few comments, but the main one that needs to be addressed is the ContactPlane header.

File: src/tribol/search/InterfacePairFinder.cpp (hunk 5 above)
Comment (srwopschall) [reply src/tribol/search/InterfacePairFinder.cpp]: We have to check the return value of the geomFilter routine to determine whether to add the interface pair to the manager or not. I moved setting pair.isContactCandidate inside geomFilter (since geomFilter is in fact responsible for performing initial checks on valid face-pairs), which simplifies the lines you point out here. I think this is cleaner?

File: src/tribol/geom/ContactPlane.hpp [hunk 6]
```diff
@@ -30,7 +30,10 @@ class ContactPlaneManager;
  * \param [in] pair interface pair containing pair related indices
  * \param [in] cMethod the Tribol contact method
  * \param [in] cCase the Tribol contact Case
- * \param [in] inContact true if pair are in contact per CG routines
+ * \param [in] isInteracting true if pair passes all computational geometry filters 
+ *
+ * \note isInteracting is true indicating a contact candidate for intersecting or 
+ *       nearly intersecting face-pairs with a positive area of overlap
```
Comment (ebchin) [inline src/tribol/geom/ContactPlane.hpp:36]: Thanks!

File: src/tribol/mesh/InterfacePairs.hpp [hunk 7]
```diff
@@ -51,11 +60,17 @@ struct InterfacePair
...
-   // boolean to hold whether pair is in contact or not in contact
-   bool inContact;
+   // boolean indicating if a binned pair is a contact candidate.
+   // A contact candidate is defined as a face-pair that is deemed geometrically proximate 
+   // by the binning coarse search, and one that passes the finer computational geometry 
+   // (CG) filter/checks. These finer checks identify face-pairs that are intersecting or 
+   // nearly intersecting with positive areas of overlap. These checks do not indicate 
+   // whether a face-pair is contacting, since the definition of 'contacting' is specific 
+   // to a particular contact method and its enforced contstraints. Rather, the CG filter
+   // identifies contact candidacy, or face-pairs likely in contact.
```
Comment (ebchin) [inline src/tribol/mesh/InterfacePairs.hpp:70]: This is helpful, too

File: src/tribol/search/InterfacePairFinder.cpp (hunk 5 above)
Comment (ebchin) [reply src/tribol/search/InterfacePairFinder.cpp]: Yes, much cleaner now.

**Review by ebchin — APPROVED** [review]
Looks good!
